In [1]:
import numpy as np
import pandas as pd
import re

## Echelle Condordance

Just to see the original concordance data (allt20.dec)

In [2]:
print(open('allt20.dec.txt').read(908))

  32 ~~~~~~~~~~~~ 00:00:00.0          +00:00:00.          19.99
  61 624-071      16:18:31.1  -0.770  +00:00:02.  -14.80  12.95
 101 HIP071363    14:35:39.66 -0.6127 +00:00:05.7 -14.703 11.700                      112 HIP071363
 101 HIP116914    23:41:55.39  1.6185 +00:00:08.5  -3.586  8.870 G0                   113 HIP116914    106 H222591      107 B-00_4553    115 HIP116914    112 HIP116914
 101 HIP033819    07:01:14.95 -0.4023 +00:00:37.9   1.037  7.620 F5V                  113 HIP033819    106 H052431      107 B+00_1760    112 HIP033819
  99 ACT0091938   08:16:57.7   1.486  +00:01:03.  -15.09   9.79            105328 9.8  86 B+00_2245     15 G113-22
 101 HIP043172    08:47:36.91  1.0103 +00:01:04.4  -9.517  7.833 G0                   113 HIP043172    107 B+00_2393    115 HIP043172    112 HIP043172
  86 ADS06366C    07:47:04.2          +00:01:06.          12.43             26 88 3.7  52 OB29C


<br>
Read the data and store it to a dataframe

In [3]:
widths = [1,3,1,12,1,11,1,7,1,11,1,7,1,6,22,3,1,12,1,3,1,12,1,3,1,12,1,3,1,12,1,3,1,12,1,3,1,12,1,3,1,12,1,3,1,12,1,3,1,12,1,3,1,12,1,3,1,12,1,3,1,12,1,3,1,12,1,3,1,12,1,3,1,12]
names = ['prog_id_1','target_name_1','ra','pm_ra','dec','pm_dec','mag','prog_id_2','target_name_2','prog_id_3','target_name_3','prog_id_4','target_name_4','prog_id_5','target_name_5','prog_id_6','target_name_6','prog_id_7','target_name_7','prog_id_8','target_name_8','prog_id_9','target_name_9','prog_id_10','target_name_10','prog_id_11','target_name_11','prog_id_12','target_name_12','prog_id_13','target_name_13','prog_id_14','target_name_14','prog_id_15','target_name_15']
usecols = [1,3,5,7,9,11,13,15,17,19,21,23,25,27,29,31,33,35,37,39,41,43,45,47,49,51,53,55,57,59,61,63,65,67,69]

In [4]:
df = pd.read_fwf('allt20.dec.txt', widths=widths, header=None, names=names, usecols=usecols, skipinitialspace=True)

In [5]:
df.head(8)

prog_id_1 target_name_1           ra   pm_ra          dec  pm_dec     mag  \
0         32  ~~~~~~~~~~~~   00:00:00.0     NaN   +00:00:00.     NaN  19.990   
1         61       624-071   16:18:31.1 -0.7700   +00:00:02. -14.800  12.950   
2        101     HIP071363  14:35:39.66 -0.6127  +00:00:05.7 -14.703  11.700   
3        101     HIP116914  23:41:55.39  1.6185  +00:00:08.5  -3.586   8.870   
4        101     HIP033819  07:01:14.95 -0.4023  +00:00:37.9   1.037   7.620   
5         99    ACT0091938   08:16:57.7  1.4860   +00:01:03. -15.090   9.790   
6        101     HIP043172  08:47:36.91  1.0103  +00:01:04.4  -9.517   7.833   
7         86     ADS06366C   07:47:04.2     NaN   +00:01:06.     NaN  12.430   

   prog_id_2 target_name_2  prog_id_3       ...       prog_id_11  \
0        NaN           NaN        NaN       ...              NaN   
1        NaN           NaN        NaN       ...              NaN   
2      112.0     HIP071363        NaN       ...              NaN   
3      113.0     HIP116914      106.0       ...              NaN   
4      113.0     HIP033819      106.0       ...              NaN   
5       86.0     B+00_2245       15.0       ...              NaN   
6      113.0     HIP043172      107.0       ...              NaN   
7       52.0         OB29C        NaN       ...              NaN   

   target_name_11 prog_id_12  target_name_12 prog_id_13  target_name_13  \
0             NaN        NaN             NaN        NaN             NaN   
1             NaN        NaN             NaN        NaN             NaN   
2             NaN        NaN             NaN        NaN             NaN   
3             NaN        NaN             NaN        NaN             NaN   
4             NaN        NaN             NaN        NaN             NaN   
5             NaN        NaN             NaN        NaN             NaN   
6             NaN        NaN             NaN        NaN             NaN   
7             NaN        NaN             NaN        NaN             NaN   

  prog_id_14  target_name_14 prog_id_15  target_name_15  
0        NaN             NaN        NaN             NaN  
1        NaN             NaN        NaN             NaN  
2        NaN             NaN        NaN             NaN  
3        NaN             NaN        NaN             NaN  
4        NaN             NaN        NaN             NaN  
5        NaN             NaN        NaN             NaN  
6        NaN             NaN        NaN             NaN  
7        NaN             NaN        NaN             NaN  

[8 rows x 35 columns]

<br>
Add running ID (target_id) to the dataframe

In [6]:
df['target_id'] = df.index + 1

In [7]:
df.head(8)

prog_id_1 target_name_1           ra   pm_ra          dec  pm_dec     mag  \
0         32  ~~~~~~~~~~~~   00:00:00.0     NaN   +00:00:00.     NaN  19.990   
1         61       624-071   16:18:31.1 -0.7700   +00:00:02. -14.800  12.950   
2        101     HIP071363  14:35:39.66 -0.6127  +00:00:05.7 -14.703  11.700   
3        101     HIP116914  23:41:55.39  1.6185  +00:00:08.5  -3.586   8.870   
4        101     HIP033819  07:01:14.95 -0.4023  +00:00:37.9   1.037   7.620   
5         99    ACT0091938   08:16:57.7  1.4860   +00:01:03. -15.090   9.790   
6        101     HIP043172  08:47:36.91  1.0103  +00:01:04.4  -9.517   7.833   
7         86     ADS06366C   07:47:04.2     NaN   +00:01:06.     NaN  12.430   

   prog_id_2 target_name_2  prog_id_3    ...     target_name_11  prog_id_12  \
0        NaN           NaN        NaN    ...                NaN         NaN   
1        NaN           NaN        NaN    ...                NaN         NaN   
2      112.0     HIP071363        NaN    ...                NaN         NaN   
3      113.0     HIP116914      106.0    ...                NaN         NaN   
4      113.0     HIP033819      106.0    ...                NaN         NaN   
5       86.0     B+00_2245       15.0    ...                NaN         NaN   
6      113.0     HIP043172      107.0    ...                NaN         NaN   
7       52.0         OB29C        NaN    ...                NaN         NaN   

  target_name_12  prog_id_13 target_name_13  prog_id_14 target_name_14  \
0            NaN         NaN            NaN         NaN            NaN   
1            NaN         NaN            NaN         NaN            NaN   
2            NaN         NaN            NaN         NaN            NaN   
3            NaN         NaN            NaN         NaN            NaN   
4            NaN         NaN            NaN         NaN            NaN   
5            NaN         NaN            NaN         NaN            NaN   
6            NaN         NaN            NaN         NaN            NaN   
7            NaN         NaN            NaN         NaN            NaN   

   prog_id_15 target_name_15  target_id  
0         NaN            NaN          1  
1         NaN            NaN          2  
2         NaN            NaN          3  
3         NaN            NaN          4  
4         NaN            NaN          5  
5         NaN            NaN          6  
6         NaN            NaN          7  
7         NaN            NaN          8  

[8 rows x 36 columns]

<br>
df_basic_info: Basic information from the concordance table. These may not be that accurate.<br>
df_basic_info.csv: csv format

In [8]:
df_basic_info = df[df.prog_id_1>0].loc[:,['target_id','ra','pm_ra','dec','pm_dec','mag']]
len(df_basic_info)

33909

In [9]:
df_basic_info.head(8)

target_id           ra   pm_ra          dec  pm_dec     mag
0          1   00:00:00.0     NaN   +00:00:00.     NaN  19.990
1          2   16:18:31.1 -0.7700   +00:00:02. -14.800  12.950
2          3  14:35:39.66 -0.6127  +00:00:05.7 -14.703  11.700
3          4  23:41:55.39  1.6185  +00:00:08.5  -3.586   8.870
4          5  07:01:14.95 -0.4023  +00:00:37.9   1.037   7.620
5          6   08:16:57.7  1.4860   +00:01:03. -15.090   9.790
6          7  08:47:36.91  1.0103  +00:01:04.4  -9.517   7.833
7          8   07:47:04.2     NaN   +00:01:06.     NaN  12.430

In [10]:
df_basic_info.to_csv('df_basic_info.csv', index=False)

<br>
Extract prog_id, target_name, and target_id on each column

In [11]:
df1 = df[df.prog_id_1>0].loc[:,['prog_id_1','target_name_1','target_id']]
df2 = df[df.prog_id_2>0].loc[:,['prog_id_2','target_name_2','target_id']]
df3 = df[df.prog_id_3>0].loc[:,['prog_id_3','target_name_3','target_id']]
df4 = df[df.prog_id_4>0].loc[:,['prog_id_4','target_name_4','target_id']]
df5 = df[df.prog_id_5>0].loc[:,['prog_id_5','target_name_5','target_id']]
df6 = df[df.prog_id_6>0].loc[:,['prog_id_6','target_name_6','target_id']]
df7 = df[df.prog_id_7>0].loc[:,['prog_id_7','target_name_7','target_id']]
df8 = df[df.prog_id_8>0].loc[:,['prog_id_8','target_name_8','target_id']]
df9 = df[df.prog_id_9>0].loc[:,['prog_id_9','target_name_9','target_id']]
df10 = df[df.prog_id_10>0].loc[:,['prog_id_10','target_name_10','target_id']]
df11 = df[df.prog_id_11>0].loc[:,['prog_id_11','target_name_11','target_id']]
df12 = df[df.prog_id_12>0].loc[:,['prog_id_12','target_name_12','target_id']]
df13 = df[df.prog_id_13>0].loc[:,['prog_id_13','target_name_13','target_id']]
df14 = df[df.prog_id_14>0].loc[:,['prog_id_14','target_name_14','target_id']]
df15 = df[df.prog_id_15>0].loc[:,['prog_id_15','target_name_15','target_id']]

In [12]:
df1.head(8)

prog_id_1 target_name_1  target_id
0         32  ~~~~~~~~~~~~          1
1         61       624-071          2
2        101     HIP071363          3
3        101     HIP116914          4
4        101     HIP033819          5
5         99    ACT0091938          6
6        101     HIP043172          7
7         86     ADS06366C          8

In [13]:
df15

prog_id_15 target_name_15  target_id
369         143.0      HIP070319        370
1380        143.0      HIP088622       1381
23048       143.0      HIP042438      23049

In [14]:
len(df1)+len(df2)+len(df3)+len(df4)+len(df5)+len(df6)+len(df7)+len(df8)+len(df9)+len(df10)+len(df11)+len(df12)+len(df13)+len(df14)+len(df15)

138602

In [15]:
# unify column names
df1.columns = ['prog_id','target_name','target_id']
df2.columns = ['prog_id','target_name','target_id']
df3.columns = ['prog_id','target_name','target_id']
df4.columns = ['prog_id','target_name','target_id']
df5.columns = ['prog_id','target_name','target_id']
df6.columns = ['prog_id','target_name','target_id']
df7.columns = ['prog_id','target_name','target_id']
df8.columns = ['prog_id','target_name','target_id']
df9.columns = ['prog_id','target_name','target_id']
df10.columns = ['prog_id','target_name','target_id']
df11.columns = ['prog_id','target_name','target_id']
df12.columns = ['prog_id','target_name','target_id']
df13.columns = ['prog_id','target_name','target_id']
df14.columns = ['prog_id','target_name','target_id']
df15.columns = ['prog_id','target_name','target_id']

In [16]:
# express prog_id in integer (not float)
df2['prog_id'] = df2['prog_id'].astype('int64')
df3['prog_id'] = df3['prog_id'].astype('int64')
df4['prog_id'] = df4['prog_id'].astype('int64')
df5['prog_id'] = df5['prog_id'].astype('int64')
df6['prog_id'] = df6['prog_id'].astype('int64')
df7['prog_id'] = df7['prog_id'].astype('int64')
df8['prog_id'] = df8['prog_id'].astype('int64')
df9['prog_id'] = df9['prog_id'].astype('int64')
df10['prog_id'] = df10['prog_id'].astype('int64')
df11['prog_id'] = df11['prog_id'].astype('int64')
df12['prog_id'] = df12['prog_id'].astype('int64')
df13['prog_id'] = df13['prog_id'].astype('int64')
df14['prog_id'] = df14['prog_id'].astype('int64')
df15['prog_id'] = df15['prog_id'].astype('int64')

In [17]:
df1.head(8)

prog_id   target_name  target_id
0       32  ~~~~~~~~~~~~          1
1       61       624-071          2
2      101     HIP071363          3
3      101     HIP116914          4
4      101     HIP033819          5
5       99    ACT0091938          6
6      101     HIP043172          7
7       86     ADS06366C          8

In [18]:
df15

prog_id target_name  target_id
369        143   HIP070319        370
1380       143   HIP088622       1381
23048      143   HIP042438      23049

<br>
df_prog_target: Concatenated 15 dataframes<br>
df_prog_target.csv: csv format, unique rows only (no duplicates)

In [19]:
df_prog_target = pd.concat([df1,df2,df3,df4,df5,df6,df7,df8,df9,df10,df11,df12,df13,df14,df15]).sort_values(['target_name','prog_id'])

In [20]:
df_prog_target.head(8)

prog_id target_name  target_id
562         2      +01755        563
2177       14      +06648       2178
7054        2      +17841       7055
7261       56     +182930       7262
8038        2      +19832       8039
9361        2      +22669       9362
9463        2      +23755       9464
9465       46     +242584       9466

In [21]:
len(df_prog_target)

138602

In [22]:
len(df_prog_target.drop_duplicates())

138600

In [23]:
# df_prog_target.csv: unique rows only
df_prog_target.drop_duplicates().to_csv('df_prog_target.csv', index=False)

In [24]:
# Just to see duplicate rows
dff_temp = df_prog_target.copy()
dff_temp['dup'] = dff_temp.duplicated(keep=False)
dff_temp[dff_temp.dup == True]

prog_id target_name  target_id   dup
3618       106    H079191A       3619  True
3618       106    H079191A       3619  True
25956      112   HIP073846      25957  True
25956      112   HIP073846      25957  True

## SIMBAD

Input: coords_raw.txt (coordinates from the concordance file)<br>
SIMBAD coordinate query - 10 arc sec (with http://simbad.u-strasbg.fr/simbad/sim-fout)<br>
Output: coords_simbad.txt<br>
<br>
Major problems<br>
(1) No object found from SIMBAD<br>
(2) Multiple objects found (6,072 out of 33,909) - use the smallest angulation for now (to begin with)<br>
(3) The same object is found for different Echelle objects (e.g., HIP013117A, HIP013117AB, HIP013117B)

In [25]:
in_file = open("coords_simbad.txt", "r")
out_file_1 = open("coords_simbad_output_1.txt", "w")
out_file_2 = open("coords_simbad_output_2.txt", "w")

i = 1
for line in in_file:
    if re.match("(.*)No astronomical object found(.*)", line):
        print >> out_file_1, i, "|", "~|~|~|~ ~ ~ ~ ~ ~|~ ~|~ ~|~|~|~|~|~|~\n",
#        print >> out_file, i, "|", "|||||||||||\n",
#        print >> out_file, i, "|", line,
        i = i + 1
    elif re.match("1(\|| \|)", line):
        print >> out_file_1, i, "|", line,
        i = i + 1
    elif re.match("[2-9](\|| \|)", line):
        print >> out_file_2, i, "|", line,

in_file.close()
out_file_1.close()
out_file_2.close()

# out_file_1 - rank 1 objects or no object found
# out_file_2 - rank 2-9 objects

 <br>
 For SIMBAD objects found and chosen to be used, polish and store the data.

In [26]:
names = ['target_id','rank','sep','simbad_identifier','coords1','coords2','pm','plx','rv','z','magB','magV','sp_type']

In [27]:
df1 = pd.read_csv('coords_simbad_output_1.txt', sep='|', header=None, names=names, skipinitialspace=True)
len(df1)

33909

In [28]:
df1.head(8)

target_id rank   sep          simbad_identifier  \
0          1    1  0.00  Fermi bn100227067           
1          2    1  5.98  LP 624-71                   
2          3    1  0.05  Wolf 535                    
3          4    1  0.11  HD 222591                   
4          5    1  0.07  HD 52431                    
5          6    1  1.37  BD+00 2245                  
6          7    1  0.07  BD+00 2393                  
7          8    1  1.80  BD+00 2079C                 

                         coords1                      coords2  \
0    00 00 00.0 +00 00 00         000.0000 +00.0000             
1    16 18 31.388 +00 00 06.14    244.630783 +00.001706         
2  14 35 39.66222 +00 00 05.6690  218.91525924 +00.00157473     
3  23 41 55.39629 +00 00 08.4486  355.48081788 +00.00234682     
4  07 01 14.94596 +00 00 37.9298  105.31227482 +00.01053604     
5    08 16 57.778 +00 01 03.72    124.240742 +00.017700         
6  08 47 36.90543 +00 01 04.3879  131.90377261 +00.01788553     
7    07 47 04.1 +00 01 05         116.7671 +00.0181             

                pm    plx         rv             z    magB    magV  \
0             ~ ~       ~  ~          ~                  ~       ~   
1        -81 -104       ~  ~          ~             13.61   12.95    
2  -88.10 -148.56   14.73  ~          ~             12.68   11.51    
3   243.00 -35.07   20.30  -9.60      -0.000032      9.52    8.79    
4    -60.41 11.16   12.30  48.90      0.000163       7.97    7.58    
5    224.4 -150.4       ~  53.81      0.000180      10.28    9.69    
6   151.03 -96.32   16.61  ~          ~              8.410   7.77    
7             ~ ~       ~  ~          ~             12.67   12.33    

           sp_type  
0  ~                
1  ~                
2  ~                
3  G8V              
4  F2V              
5  G0               
6  G0               
7  F5IV

In [29]:
df1['rah'],df1['ram'],df1['ras'],df1['decd'],df1['decm'],df1['decs'] = df1['coords1'].str.split(' ',5).str
df1['ra_d'],df1['dec_d'] = df1['coords2'].str.split(' ',1).str
df1['pm_ra'],df1['pm_dec'] = df1['pm'].str.split(' ',1).str

In [30]:
df1.head(8)

target_id rank   sep          simbad_identifier  \
0          1    1  0.00  Fermi bn100227067           
1          2    1  5.98  LP 624-71                   
2          3    1  0.05  Wolf 535                    
3          4    1  0.11  HD 222591                   
4          5    1  0.07  HD 52431                    
5          6    1  1.37  BD+00 2245                  
6          7    1  0.07  BD+00 2393                  
7          8    1  1.80  BD+00 2079C                 

                         coords1                      coords2  \
0    00 00 00.0 +00 00 00         000.0000 +00.0000             
1    16 18 31.388 +00 00 06.14    244.630783 +00.001706         
2  14 35 39.66222 +00 00 05.6690  218.91525924 +00.00157473     
3  23 41 55.39629 +00 00 08.4486  355.48081788 +00.00234682     
4  07 01 14.94596 +00 00 37.9298  105.31227482 +00.01053604     
5    08 16 57.778 +00 01 03.72    124.240742 +00.017700         
6  08 47 36.90543 +00 01 04.3879  131.90377261 +00.01788553     
7    07 47 04.1 +00 01 05         116.7671 +00.0181             

                pm    plx         rv             z    ...    rah ram  \
0             ~ ~       ~  ~          ~               ...     00  00   
1        -81 -104       ~  ~          ~               ...     16  18   
2  -88.10 -148.56   14.73  ~          ~               ...     14  35   
3   243.00 -35.07   20.30  -9.60      -0.000032       ...     23  41   
4    -60.41 11.16   12.30  48.90      0.000163        ...     07  01   
5    224.4 -150.4       ~  53.81      0.000180        ...     08  16   
6   151.03 -96.32   16.61  ~          ~               ...     08  47   
7             ~ ~       ~  ~          ~               ...     07  47   

        ras decd decm       decs          ra_d               dec_d   pm_ra  \
0      00.0  +00   00  00             000.0000  +00.0000                 ~   
1    31.388  +00   00    06.14      244.630783    +00.001706           -81   
2  39.66222  +00   00    05.6690  218.91525924      +00.00157473    -88.10   
3  55.39629  +00   00    08.4486  355.48081788      +00.00234682    243.00   
4  14.94596  +00   00    37.9298  105.31227482      +00.01053604    -60.41   
5    57.778  +00   01    03.72      124.240742    +00.017700         224.4   
6  36.90543  +00   01    04.3879  131.90377261      +00.01788553    151.03   
7      04.1  +00   01  05             116.7671  +00.0181                 ~   

     pm_dec  
0        ~   
1     -104   
2  -148.56   
3   -35.07   
4    11.16   
5   -150.4   
6   -96.32   
7        ~   

[8 rows x 23 columns]

In [31]:
df1['ra'] = df1['rah'].astype(str).str.cat(df1['ram'].astype(str), sep=':')
df1['ra'] = df1['ra'].astype(str).str.cat(df1['ras'].astype(str), sep=':')

df1['dec'] = df1['decd'].astype(str).str.cat(df1['decm'].astype(str), sep=':')
df1['dec'] = df1['dec'].astype(str).str.cat(df1['decs'].astype(str), sep=':')

In [32]:
df1.head(8)

target_id rank   sep          simbad_identifier  \
0          1    1  0.00  Fermi bn100227067           
1          2    1  5.98  LP 624-71                   
2          3    1  0.05  Wolf 535                    
3          4    1  0.11  HD 222591                   
4          5    1  0.07  HD 52431                    
5          6    1  1.37  BD+00 2245                  
6          7    1  0.07  BD+00 2393                  
7          8    1  1.80  BD+00 2079C                 

                         coords1                      coords2  \
0    00 00 00.0 +00 00 00         000.0000 +00.0000             
1    16 18 31.388 +00 00 06.14    244.630783 +00.001706         
2  14 35 39.66222 +00 00 05.6690  218.91525924 +00.00157473     
3  23 41 55.39629 +00 00 08.4486  355.48081788 +00.00234682     
4  07 01 14.94596 +00 00 37.9298  105.31227482 +00.01053604     
5    08 16 57.778 +00 01 03.72    124.240742 +00.017700         
6  08 47 36.90543 +00 01 04.3879  131.90377261 +00.01788553     
7    07 47 04.1 +00 01 05         116.7671 +00.0181             

                pm    plx         rv             z        ...         \
0             ~ ~       ~  ~          ~                   ...          
1        -81 -104       ~  ~          ~                   ...          
2  -88.10 -148.56   14.73  ~          ~                   ...          
3   243.00 -35.07   20.30  -9.60      -0.000032           ...          
4    -60.41 11.16   12.30  48.90      0.000163            ...          
5    224.4 -150.4       ~  53.81      0.000180            ...          
6   151.03 -96.32   16.61  ~          ~                   ...          
7             ~ ~       ~  ~          ~                   ...          

        ras decd decm       decs          ra_d               dec_d   pm_ra  \
0      00.0  +00   00  00             000.0000  +00.0000                 ~   
1    31.388  +00   00    06.14      244.630783    +00.001706           -81   
2  39.66222  +00   00    05.6690  218.91525924      +00.00157473    -88.10   
3  55.39629  +00   00    08.4486  355.48081788      +00.00234682    243.00   
4  14.94596  +00   00    37.9298  105.31227482      +00.01053604    -60.41   
5    57.778  +00   01    03.72      124.240742    +00.017700         224.4   
6  36.90543  +00   01    04.3879  131.90377261      +00.01788553    151.03   
7      04.1  +00   01  05             116.7671  +00.0181                 ~   

     pm_dec              ra               dec  
0        ~       00:00:00.0  +00:00:00         
1     -104     16:18:31.388    +00:00:06.14    
2  -148.56   14:35:39.66222    +00:00:05.6690  
3   -35.07   23:41:55.39629    +00:00:08.4486  
4    11.16   07:01:14.94596    +00:00:37.9298  
5   -150.4     08:16:57.778    +00:01:03.72    
6   -96.32   08:47:36.90543    +00:01:04.3879  
7        ~       07:47:04.1  +00:01:05         

[8 rows x 25 columns]

In [33]:
df1.drop(['coords1','coords2','pm','rah','ram','ras','decd','decm','decs'], axis = 1).head(8)

target_id rank   sep          simbad_identifier    plx         rv  \
0          1    1  0.00  Fermi bn100227067              ~  ~           
1          2    1  5.98  LP 624-71                      ~  ~           
2          3    1  0.05  Wolf 535                   14.73  ~           
3          4    1  0.11  HD 222591                  20.30  -9.60       
4          5    1  0.07  HD 52431                   12.30  48.90       
5          6    1  1.37  BD+00 2245                     ~  53.81       
6          7    1  0.07  BD+00 2393                 16.61  ~           
7          8    1  1.80  BD+00 2079C                    ~  ~           

              z    magB    magV          sp_type          ra_d  \
0  ~                  ~       ~  ~                    000.0000   
1  ~             13.61   12.95   ~                  244.630783   
2  ~             12.68   11.51   ~                218.91525924   
3  -0.000032      9.52    8.79   G8V              355.48081788   
4  0.000163       7.97    7.58   F2V              105.31227482   
5  0.000180      10.28    9.69   G0                 124.240742   
6  ~              8.410   7.77   G0               131.90377261   
7  ~             12.67   12.33   F5IV                 116.7671   

                dec_d   pm_ra    pm_dec              ra               dec  
0  +00.0000                 ~        ~       00:00:00.0  +00:00:00         
1    +00.001706           -81     -104     16:18:31.388    +00:00:06.14    
2      +00.00157473    -88.10  -148.56   14:35:39.66222    +00:00:05.6690  
3      +00.00234682    243.00   -35.07   23:41:55.39629    +00:00:08.4486  
4      +00.01053604    -60.41    11.16   07:01:14.94596    +00:00:37.9298  
5    +00.017700         224.4   -150.4     08:16:57.778    +00:01:03.72    
6      +00.01788553    151.03   -96.32   08:47:36.90543    +00:01:04.3879  
7  +00.0181                 ~        ~       07:47:04.1  +00:01:05

In [34]:
df1 = df1[['target_id','rank','sep','simbad_identifier','ra','pm_ra','dec','pm_dec','ra_d','dec_d','plx','rv','z','magB','magV','sp_type']]

In [35]:
df1.head(8)

target_id rank   sep          simbad_identifier              ra   pm_ra  \
0          1    1  0.00  Fermi bn100227067              00:00:00.0       ~   
1          2    1  5.98  LP 624-71                    16:18:31.388     -81   
2          3    1  0.05  Wolf 535                   14:35:39.66222  -88.10   
3          4    1  0.11  HD 222591                  23:41:55.39629  243.00   
4          5    1  0.07  HD 52431                   07:01:14.94596  -60.41   
5          6    1  1.37  BD+00 2245                   08:16:57.778   224.4   
6          7    1  0.07  BD+00 2393                 08:47:36.90543  151.03   
7          8    1  1.80  BD+00 2079C                    07:47:04.1       ~   

                dec    pm_dec          ra_d               dec_d    plx  \
0  +00:00:00               ~       000.0000  +00.0000                ~   
1    +00:00:06.14       -104     244.630783    +00.001706            ~   
2    +00:00:05.6690  -148.56   218.91525924      +00.00157473    14.73   
3    +00:00:08.4486   -35.07   355.48081788      +00.00234682    20.30   
4    +00:00:37.9298    11.16   105.31227482      +00.01053604    12.30   
5    +00:01:03.72     -150.4     124.240742    +00.017700            ~   
6    +00:01:04.3879   -96.32   131.90377261      +00.01788553    16.61   
7  +00:01:05               ~       116.7671  +00.0181                ~   

          rv             z    magB    magV          sp_type  
0  ~          ~                  ~       ~  ~                
1  ~          ~             13.61   12.95   ~                
2  ~          ~             12.68   11.51   ~                
3  -9.60      -0.000032      9.52    8.79   G8V              
4  48.90      0.000163       7.97    7.58   F2V              
5  53.81      0.000180      10.28    9.69   G0               
6  ~          ~              8.410   7.77   G0               
7  ~          ~             12.67   12.33   F5IV

In [36]:
# remove trailing whitespaces
df1.loc[:,('simbad_identifier')] = df1.loc[:,('simbad_identifier')].str.strip()
df1.loc[:,('pm_ra')] = df1.loc[:,('pm_ra')].str.strip()
df1.loc[:,('dec')] = df1.loc[:,('dec')].str.strip()
df1.loc[:,('pm_dec')] = df1.loc[:,('pm_dec')].str.strip()
df1.loc[:,('dec_d')] = df1.loc[:,('dec_d')].str.strip()
df1.loc[:,('rv')] = df1.loc[:,('rv')].str.strip()
df1.loc[:,('z')] = df1.loc[:,('z')].str.strip()
df1.loc[:,('magB')] = df1.loc[:,('magB')].str.strip()
df1.loc[:,('magV')] = df1.loc[:,('magV')].str.strip()
df1.loc[:,('sp_type')] = df1.loc[:,('sp_type')].str.strip()

In [37]:
df1.head(8)

target_id rank   sep  simbad_identifier              ra   pm_ra  \
0          1    1  0.00  Fermi bn100227067      00:00:00.0       ~   
1          2    1  5.98          LP 624-71    16:18:31.388     -81   
2          3    1  0.05           Wolf 535  14:35:39.66222  -88.10   
3          4    1  0.11          HD 222591  23:41:55.39629  243.00   
4          5    1  0.07           HD 52431  07:01:14.94596  -60.41   
5          6    1  1.37         BD+00 2245    08:16:57.778   224.4   
6          7    1  0.07         BD+00 2393  08:47:36.90543  151.03   
7          8    1  1.80        BD+00 2079C      07:47:04.1       ~   

              dec   pm_dec          ra_d         dec_d    plx     rv  \
0       +00:00:00        ~      000.0000      +00.0000      ~      ~   
1    +00:00:06.14     -104    244.630783    +00.001706      ~      ~   
2  +00:00:05.6690  -148.56  218.91525924  +00.00157473  14.73      ~   
3  +00:00:08.4486   -35.07  355.48081788  +00.00234682  20.30  -9.60   
4  +00:00:37.9298    11.16  105.31227482  +00.01053604  12.30  48.90   
5    +00:01:03.72   -150.4    124.240742    +00.017700      ~  53.81   
6  +00:01:04.3879   -96.32  131.90377261  +00.01788553  16.61      ~   
7       +00:01:05        ~      116.7671      +00.0181      ~      ~   

           z   magB   magV sp_type  
0          ~      ~      ~       ~  
1          ~  13.61  12.95       ~  
2          ~  12.68  11.51       ~  
3  -0.000032   9.52   8.79     G8V  
4   0.000163   7.97   7.58     F2V  
5   0.000180  10.28   9.69      G0  
6          ~  8.410   7.77      G0  
7          ~  12.67  12.33    F5IV

In [38]:
# df_simbad_selected.csv - SIMBAD objects found and chosen to be used
df1.to_csv('df_simbad_selected.csv', index=False)

<br>
Repeat the same job for SIMBAD objects found and not chosen to be used

In [39]:
df2 = pd.read_csv('coords_simbad_output_2.txt', sep='|', header=None, names=names, skipinitialspace=True)

df2['rah'],df2['ram'],df2['ras'],df2['decd'],df2['decm'],df2['decs'] = df2['coords1'].str.split(' ',5).str
df2['ra_d'],df2['dec_d'] = df2['coords2'].str.split(' ',1).str
df2['pm_ra'],df2['pm_dec'] = df2['pm'].str.split(' ',1).str

df2['ra'] = df2['rah'].astype(str).str.cat(df2['ram'].astype(str), sep=':')
df2['ra'] = df2['ra'].astype(str).str.cat(df2['ras'].astype(str), sep=':')

df2['dec'] = df2['decd'].astype(str).str.cat(df2['decm'].astype(str), sep=':')
df2['dec'] = df2['dec'].astype(str).str.cat(df2['decs'].astype(str), sep=':')

df2.drop(['coords1','coords2','pm','rah','ram','ras','decd','decm','decs'], axis = 1)

df2 = df2[['target_id','rank','sep','simbad_identifier','ra','pm_ra','dec','pm_dec','ra_d','dec_d','plx','rv','z','magB','magV','sp_type']]

df2.head(8)

target_id  rank   sep          simbad_identifier              ra   pm_ra  \
0          5     2  8.26  2MASS J23415511+0000013       23:41:55.12     244   
1         38     2  0.03  HD 164509                  18:01:31.22790   -7.40   
2         40     2  0.58  ** GIC 122                     14:52:30.6       ~   
3         49     2  8.50  SSV LDN 1630 15                05:46:35.5       ~   
4         54     2  0.06  HD 167278                  18:14:33.65346   17.98   
5         54     3  1.29  HD 167278B                   18:14:33.736    19.5   
6         55     2  0.56  PPM 700433                   12:17:15.380  -147.0   
7         62     2  4.62  * d Ser                    18:27:12.50775   -3.67   

                  dec   pm_dec          ra_d               dec_d    plx  \
0   +00:00:01.3           -37      355.47967   +00.00036              ~   
1      +00:06:16.3990  -20.98   270.38011626      +00.10455527    19.07   
2  +00:06:37                ~       223.1275  +00.1103                ~   
3    +00:08:51              ~       086.6479  +00.1475                ~   
4      +00:10:32.9342  -31.02   273.64022276      +00.17581504     8.74   
5    +00:10:32.81       -35.3     273.640567    +00.175781            ~   
6    +00:10:38.06      -164.0     184.314083    +00.177239            ~   
7      +00:11:45.9912   -8.61   276.80211561      +00.19610865     6.95   

          rv             z   magB    magV          sp_type  
0  ~          ~                 ~       ~  ~                
1  13.63      0.000045      8.723   8.103  G2V              
2  ~          ~                 ~       ~  ~                
3  ~          ~                 ~       ~  ~                
4  -14.70     -0.000049     8.08    7.67   F5V              
5  ~          ~             8.94    8.49   ~                
6  ~          ~                 ~  10.5    ~                
7  -23.3      -0.000078     5.71    5.21   A0Vs+G:III

In [40]:
len(df2['target_id'].unique())

6072

In [41]:
# remove trailing whitespaces
df2.loc[:,('simbad_identifier')] = df2.loc[:,('simbad_identifier')].str.strip()
df2.loc[:,('pm_ra')] = df2.loc[:,('pm_ra')].str.strip()
df2.loc[:,('dec')] = df2.loc[:,('dec')].str.strip()
df2.loc[:,('pm_dec')] = df2.loc[:,('pm_dec')].str.strip()
df2.loc[:,('dec_d')] = df2.loc[:,('dec_d')].str.strip()
df2.loc[:,('rv')] = df2.loc[:,('rv')].str.strip()
df2.loc[:,('z')] = df2.loc[:,('z')].str.strip()
df2.loc[:,('magB')] = df2.loc[:,('magB')].str.strip()
df2.loc[:,('magV')] = df2.loc[:,('magV')].str.strip()
df2.loc[:,('sp_type')] = df2.loc[:,('sp_type')].str.strip()

In [42]:
# df_simbad_unselected.csv - SIMBAD objects found but not chosen to be used
df2.to_csv('df_simbad_unselected.csv', index=False)

<br>
<br>
All the following tables can be joined by target_id.<br>
<br>
df_basic_info.csv<br>
basic information from the concordance table. These may not be that accurate.<br>
<br>
df_prog_target.csv<br>
unique rows only, sorted by target_name and then prog_id<br>
<br>
df_simbad_selected.csv<br>
SIMBAD objects found and chosen to be used<br>
<br>
df_simbad_unselected.csv<br>
SIMBAD objects found but not chosen to be used